src/database/populate.py

```
@@ -57,34 +57,35 @@ def _preprocess_csv(self) -> pd.DataFrame:
         :return: A Pandas DataFrame containing cleaned movie data.
         """
         data = pd.read_csv(self._csv_file_path)
-        data = data.drop_duplicates(subset=['names', 'date_x'], keep='first')
+        data = data.drop_duplicates(subset=["names", "date_x"], keep="first")
```

Check the drop_duplicates call and the column names used here. The code uses `['names','date_x']`; ensure the CSV always has a `names` column (not `name`) and `date_x` as expected. A mismatch here will silently change behavior or raise KeyError. If the CSV column names can vary, validate presence and fail fast with a helpful message.